File: src/main/java/ru/dushkinmir/absolutelyRandom/actions/Action.java
package ru.dushkinmir.absolutelyRandom.actions;

import org.bukkit.plugin.Plugin;

public abstract class Action {
    private final String name;

    protected Action(String name) {
        this.name = name;
    }

    public String getName() {
        return name.toLowerCase();
    }

    public abstract void execute(Plugin plugin);
}

File: src/main/java/ru/dushkinmir/absolutelyRandom/features/actions/Action.kt
package ru.dushkinmir.absolutelyRandom.features.actions

import org.bukkit.plugin.Plugin
import ru.dushkinmir.absolutelyRandom.core.PlayerData
import ru.dushkinmir.absolutelyRandom.core.PlayerDataManager
import java.util.*

abstract class Action(internal val name: String) {
    abstract fun execute(plugin: Plugin)

    protected fun getPlayerData(playerUUID: UUID): PlayerData {
        val actionName = this::class.simpleName!!
        return PlayerDataManager.getPlayerDataForAction(actionName, playerUUID)
    }

    protected fun removePlayerData(playerUUID: UUID) {
        val actionName = this::class.simpleName!!
        PlayerDataManager.removePlayerDataForAction(actionName, playerUUID)
    }
}
